clarifai/client/dataset.py

```diff
@@ -162,56 +172,69 @@ def _delete_failed_inputs(self, batch_input_ids: List[int]) -> Tuple[List[int],
+      self, batch_input_ids: List[int], task: str, chunk_size: int,
+      dataset_obj: ClarifaiDatasetType) -> Tuple[List[int], List[resources_pb2.Annotation]]:
     """Uploads batch of inputs and annotations concurrently to clarifai platform dataset.
     Args:
       batch_input_ids: batch input ids
+      task: Dataset type
+      chunk_size: chunk size for input, annotation protos
+      dataset_obj: ClarifaiDataset object
     Returns:
       failed_input_ids: failed input ids
       retry_annot_protos: failed annot protos
     """
-    input_protos, _ = self.dataset_obj.get_protos(batch_input_ids)
+    input_protos, _ = dataset_obj.get_protos(batch_input_ids)
     input_job_id = self._upload_inputs(input_protos)
     retry_annot_protos = []
 
     self._wait_for_inputs(input_job_id)
-    success_input_ids, failed_input_ids = self._delete_failed_inputs(batch_input_ids)
+    success_input_ids, failed_input_ids = self._delete_failed_inputs(batch_input_ids, dataset_obj)
 
-    if self.task in ["visual_detection", "visual_segmentation"]:
-      _, annotation_protos = self.dataset_obj.get_protos(success_input_ids)
-      chunked_annotation_protos = Chunker(annotation_protos, self.chunk_size).chunk()
+    if task in ["visual_detection", "visual_segmentation"]:
+      _, annotation_protos = dataset_obj.get_protos(success_input_ids)
+      chunked_annotation_protos = Chunker(annotation_protos, chunk_size).chunk()
       retry_annot_protos.extend(self._concurrent_annot_upload(chunked_annotation_protos))
 
     return failed_input_ids, retry_annot_protos
 
   def _retry_uploads(self, failed_input_ids: List[int],
-                     retry_annot_protos: List[resources_pb2.Annotation]) -> None:
+                     retry_annot_protos: List[resources_pb2.Annotation], task: str,
+                     chunk_size: int, dataset_obj: ClarifaiDatasetType) -> None:
     """Retry failed uploads.
     Args:
       failed_input_ids: failed input ids
       retry_annot_protos: failed annot protos
+      task: Dataset type
+      chunk_size: chunk size for input, annotation protos
+      dataset_obj: ClarifaiDataset object
     """
     if failed_input_ids:
-      self._upload_inputs_annotations(failed_input_ids)
+      self._upload_inputs_annotations(failed_input_ids, task, chunk_size, dataset_obj)
     if retry_annot_protos:
-      chunked_annotation_protos = Chunker(retry_annot_protos, self.chunk_size).chunk()
+      chunked_annotation_protos = Chunker(retry_annot_protos, chunk_size).chunk()
       _ = self._concurrent_annot_upload(chunked_annotation_protos)
 
-  def _data_upload(self) -> None:
-    """Uploads inputs and annotations to clarifai platform dataset."""
-    input_ids = list(range(len(self.dataset_obj)))
-    chunk_input_ids = Chunker(input_ids, self.chunk_size).chunk()
+  def _data_upload(self, task: str, chunk_size: int, dataset_obj: ClarifaiDatasetType) -> None:
```

Comment:
I guess `task` and `chunksize` should remain as part of the constructor because they are clearly used by most methods. Manually redefining them for each method seems redundant.

Reply:
I see they are method params for all methods except `delete_failed_inputs()`. They can be defined once and accessed via `self` to reduce redundancy.